DentOS_Framework/DentOsTestbed/src/dent_os_testbed/utils/test_utils/tb_utils.py

```diff
@@ -551,3 +552,23 @@ async def tb_device_tcpdump(device, interface, options, count_only=False, timeou
 
     else:
         return out
+
+
+async def tb_get_qualified_ports(device, ports, speed, duplex):
+
+    # Check dut media mode name. If it supports the speed parameter
+    if device.media_mode == "mixed":
+        raise Exception("Mixed mode is not supported")
+    if device.media_mode == "copper" and speed == 10000:
+        pytest.skip(reason=f"Can not run test in device {device.host_name}  with the speed: {speed}")
+
+    # Check that there are at least minimum 2 ports with the provided speed parameter
+    speed_ports = {}
+    for port in ports:
+        out = await Ethtool.show(input_data=[{device.host_name: [{"devname": port}]}],  parse_output=True)
+        supported_speeds = '{}baseT/{}' if "TP" in out[0][device.host_name]['parsed_output']["supported_ports"] else '{}baseSR/{}'
+        if supported_speeds.format(speed, duplex.capitalize()) in out[0][device.host_name]['result']:
+            speed_ports[port] = {"speed": speed,
+                                 "duplex": duplex}
+    assert len(speed_ports) >= 2, f"Need 2 ports with the same speed of {speed} and duplex {duplex}"
```

Comment:
Add a parameter like "required_ports=None" and if it is set, perform an assertion here